cypress/integration/student_tasks.js

```diff
@@ -205,140 +92,515 @@ describe.skip('Student tasks test', () => {
-                                        expect(status).to.equal(200);
-                                    })
-                                    .then(() => {
-                                        _selectFilter(filter);
-
-                                        cy.wait('@studentTasksData')
-                                            .should(d => {
-                                                expect(d.status).to.equal(200);
-                                            })
-                                            .and(() => {
-                                                cy.get('#' + currentTasksTree.id)
-                                                    .should(($tasksTree) => {
-                                                        expect(
-                                                            $tasksTree.find('li.slate-tasktree-item').length
-                                                        ).to.equal(length)
-                                                    });
-                                            });
-                                    });
-                            }
-                        });
-                });
-        });
-    });
+    it('Submitted Tasks Filter', ()=>{
+        _visitDashboardAsTeacher();
+        _setupTests()
+        .then(({ _deselectAllFilters, _selectFilter, _clickFilterButton, currentTasksTree }) => {
+            _clickFilterButton()
+            .then(_deselectAllFilters)
+            .then(() => _selectFilter('Submitted Tasks'))
+            .then(({ response }) => {
+                cy.wait(500); // allow dom re-render
+                cy.get('#' + currentTasksTree.id)
+                    .find('ul.slate-tasktree-list')
+                    .children('li.slate-tasktree-item')
+                    .each(($item) => {
+                        cy.get($item).find('.slate-tasktree-status').contains('Submitted'); // confirm the status text exists
+                    });
+            })
+        })
+    })
+
+    it('Completed Tasks Filter', ()=>{
+        _visitDashboardAsTeacher();
+        _setupTests()
+        .then(({ _deselectAllFilters, _selectFilter, _clickFilterButton, currentTasksTree }) => {
+            _clickFilterButton()
+            .then(_deselectAllFilters)
+            .then(() => _selectFilter('Completed Tasks'))
+            .then(({ response }) => {
+                cy.wait(500); // allow dom re-render
+                cy.get('#' + currentTasksTree.id)
+                    .find('ul.slate-tasktree-list')
+                    .children('li.slate-tasktree-item')
+                    .each(($item) => {
+                        cy.get($item).find('.slate-tasktree-status').contains('Completed'); // confirm the status text exists
+                    });
+            })
+        })
+    })
+
+    //this test could be shorter if we make an API call and test the data instead of clicking on every li element
```

Comment:
Im clicking on every Li element in order to get the full date in this formation `YYYY-MM-DD`. Im not sure how you were able to use the variable date by doing this. I had issues doing this because the date variable was not in the correct format to get passed into a dayjs method.
```let today = dayjs(),
       date = dayjs($date.text()),
       isAfter = today.isAfter(date, 'day') || today.isSame(date, 'day');
```